src/tagging/sqlite/tagging.py

```diff
@@ -0,0 +1,971 @@
+                          "database. Exception: {}. Stopping tagging " \
+                          "service agent".format(e.args)
+        table_name = ""
+        try:
+            table_name = self.tags_table
+            if self.tags_table in table_names:
+                _log.info("{} table exists. Assuming initial values have been "
+                          "loaded".format(table_name))
+            else:
+                self._init_tags()
+
+            table_name = self.tag_refs_table
+            if self.tag_refs_table in table_names:
+                _log.info("{} table exists. Assuming initial values have been "
+                          "loaded".format(table_name))
+            else:
+                self._init_tag_refs()
+
+            table_name = self.topic_tags_table
+            if self.topic_tags_table in table_names:
+                _log.info("{} table exists. Assuming initial values "
+                          "have been loaded".format(table_name))
+            else:
+                self._init_topic_tags()
+
+            table_name = self.categories_table
+            if self.categories_table in table_names:
+                _log.info("{} table exists. Assuming initial values "
+                          "have been loaded".format(table_name))
+            else:
+                self._init_categories()
+
+            table_name = self.category_tags_table
+            if self.category_tags_table in table_names:
+                _log.info("{} table exists. Assuming initial values "
+                          "have been loaded".format(table_name))
+            else:
+                self._init_category_tags()
+
+        except Exception as e:
+            err_message = "Initialization of " + table_name + \
+                          " table failed with exception: {}" \
+                          "Stopping tagging service agent. ".format(str(e))
+        if err_message:
+            _log.error(err_message)
+            self.vip.health.set_status(STATUS_BAD,
+                                       "Initialization of tagging service "
+                                       "failed")
+            status = Status.from_json(self.vip.health.get_status_json())
+            self.vip.health.send_alert(TAGGING_AGENT_SETUP_FAILED, status)
+            self.core.stop()
+
+    def load_valid_tags(self):
+        # Now cache list of tags and kind/type for validation during insert
+        cursor = self.sqlite_utils.select(
+            "SELECT name, kind from " + self.tags_table, fetch_all=False)
+        for record in cursor:
+            self.valid_tags[record[0]] = record[1]
+
+    def load_tag_refs(self):
+        # Now cache ref tags and its parent
+        cursor = self.sqlite_utils.select(
+            "SELECT tag, parent from " + self.tag_refs_table, fetch_all=False)
+        for record in cursor:
+            self.tag_refs[record[0]] = record[1]
+
+    def _init_tags(self):
+        file_path = self.resource_sub_dir + '/tags.csv'
```

Comment:
instead of using self you might use super() to say it's not this object, but the parent we are getting this from?

Reply:
Though this is not a requirement.

Reply:
I assume this could be overridden by implementing classes.

Reply:
true...and I was looking for it in the class and didn't find it :) then I remembered it was a subclass.